### PR TITLE
generatecss.php generates .min.css

### PR DESCRIPTION
**Pull request by ghost**
In this PR:
- Made the generatecss.php also generate .min.css files
- Tweaked the generatecss.php to improve output
- Re-compiled the css files
- Added the .min.css files
- Made the templates/files use the change the.min.css files (except when in DEBUG mode)

Changes I made to the generatecss.php:
- Made the function style values have spaces after commas, like `rgb(2555, 255, 255, 1)`
- Expands # rgb colours to the full 6 digit format, so `#fc9` becomes `#ffcc99`

### CODE REVIEW

**Comment by dgt41** [comment]
@nonumber Can you also make the changes for minified to 
administrator/templates/isis/component.php
administrator/templates/isis/login.php
 templates/protostar/component.php
templates/protostar/error.php

**Comment by nonumber** [comment]
PS: because of the licensing issue with the oyejorge/less.php, we have to stick with the leafo/lessphp, even though development has stopped on it.
So unfortunately this means the (non-minified) css files don't have the comments in them.
leafo/lessphp does a bad job of adding the comments from less files to the css files (all are placed at top of file). I haven't been able to quick-fix that.

**Comment by nonumber** [comment]
@dgt41 Yep, done. Thanks for reminding me. Forgot about those.

**Comment by Bakual** [comment]
According to the changelog on http://leafo.net/lessphp/changelog.html

> version 0.4.0 – August 09 2013
>
> Warning: this release has some breaking changes. If you the old selector expression syntax (eg. Bootstrap 2.3) then it’s recommended to use 0.3.9.

I think that was the reason why we never updated beyond 0.3.9.
Not sure what the impact really is.

**Comment by nonumber** [comment]
Output via latest (v0.5.0) seems fine.
Latest commits to v0.5.0 are from Apr 27, 2014

**Comment by wilsonge** [comment]
@nonumber I'm sorry but we will not be accepting PR's that make changes to the code imported by composer. If you feel these changes are essential then we can make them to the Joomla wrapper code for less (https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/less/less.php). But to be honest in these cases I'd just leave it. You're not gaining anything for the pain you're going to self-inflict trying to update the less library in the future.

**Comment by nonumber** [comment]
The leafo less library is no longer under development, so it won't ever need to get updated.
So why use composer for that library anyway? Might as well fork the whole thing and do with it what we want.

**Comment by wilsonge** [comment]
It's no longer under development but the last change was done less than 4 months ago. So we might still update it yet - it's to soon to say.

**Comment by nonumber** [comment]
I'll revert it back to the old version and do some after-process fixes in the generatecss.php.

**Comment by wilsonge** [comment]
keeping it at 0.5 is fine. it's just the changes from there you need to revert. But thanks :)

**Comment by nonumber** [comment]
v0.5.0 was not fine. Found issues with it (not inheriting overwritten variables)

**Comment by nonumber** [comment]
All should be good for testing...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6451">issues.joomla.org/joomla-cms/6451</a>.</sub>

**Comment by zero-24** [comment]
@nonumber 

>All should be good for testing...

hard to tell but we have again some conflicts. Can you fix it?

**Comment by nonumber** [comment]
Ok, redid some stuff as it was completely out of sync due to css commits.

File: templates/beez3/css/turq.less [hunk 1]
```diff
@@ -1,427 +0,0 @@
-@basecolor 		: rgb(45,53,62);
```
Comment (wilsonge) [inline templates/beez3/css/turq.less]: Why are we deleting this file?

Comment (nonumber) [inline templates/beez3/css/turq.less]: It seems to be unused.
It is not triggered in the generatecss.php and it is the only less file in the beez3 folder. All other stylesheets are plain css. So I think this just got added by someone playing around with less.

Comment (nonumber) [inline templates/beez3/css/turq.less]: Ah, it is in the generatecss.php. But seems weird that it is the only less file there.

Comment (nonumber) [inline templates/beez3/css/turq.less]: Ok, placed it back and added it to the generatecss.php

**Comment by nonumber** [comment]
Redid PR: https://github.com/joomla/joomla-cms/pull/6979

**Pull request closed without merging**